src/flp/types/fixedpoint_l2.rs

```diff
@@ -323,21 +346,70 @@ where
             gadget1_chunk_len,
         })
     }
+
+    /// This noising function can be called on the aggregation result to make
+    /// the entire aggregation process differentially private. The noise is
+    /// calibrated to result in a guarantee of `1/2 * epsilon^2` zero-concentrated
+    /// differential privacy, where `epsilon` is given by `dp_strategy.budget`.
+    fn add_noise<R: Rng>(
+        &self,
+        dp_strategy: &ZCdpDiscreteGaussian,
+        agg_result: &mut [Field128],
+        rng: &mut R,
+    ) -> Result<(), FlpError> {
+        // generate and add discrete gaussian noise for each entry
+
+        // 0. compute sensitivity of aggregation, namely 2^n
+        let sensitivity = BigUint::from(2u128).pow(self.bits_per_entry as u32);
```

Comment:
Can rustc infer this type, i.e., does this work?
```suggestion
        let sensitivity = BigUint::from(2).pow(self.bits_per_entry as u32);
```

Reply:
Nope.